src/core/view/SetsquareView.cpp

```diff
@@ -45,6 +46,9 @@ SetsquareView::~SetsquareView() noexcept { this->unregisterFromPool(); };
 void SetsquareView::on(FlagDirtyRegionRequest, const Range& rg) { this->parent->flagDirtyRegion(rg); }
 
 void SetsquareView::on(UpdateValuesRequest, double h, double rot, cairo_matrix_t m) {
+    if (h <= 0.) {
+        g_warning("Non-positive setsquare height: %f", h);
+    }
```

Comment:
Does this actually happen? Should this be asserted?

Reply:
No, I don't think it ever happens. I changed it to an assertion.